ui/admin/app/components/form/worker/create-worker-led/index.js

```diff
@@ -1,22 +1,48 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
 
 export default class FormWorkerCreateWorkerLedComponent extends Component {
-  // =attributes
+  // =services
+  @service features;
 
+  // =attributes
   @tracked generatedWorkerAuthToken;
   @tracked clusterId;
   @tracked ipAddress;
   @tracked configFilePath;
   @tracked workerTags;
+  @tracked initialUpstreams;
+
+  // =methods
+
+  textFilter(val) {
+    return val
+      .split(',')
+      .filter((itm) => itm.trim())
+      .map((itm) => `"${itm?.trim()}"`)
+      .join(', ');
+  }
 
+  // =properties
+
+  /**
+   * Returns directory creation text for `<Rose::CodeEditor>`.
+   * @type {string}
+   */
   get createConfigText() {
     return `mkdir ${this.configFilePath || '<path>'}/ ;\\
 touch ${this.configFilePath || '<path>'}/pki-worker.hcl`;
   }
 
+  /**
+   * Returns config creation text for `<Rose::CodeEditor>`. The user will see
+   * different outputs based on if they are using `hcp` or `oss` binaries.
+   * @type {string}
+   */
   get workerConfigText() {
-    return `hcp_boundary_cluster_id = "${this.clusterId || '<config_id>'}"
+    if (this.features.isEnabled('byow-pki-hcp-cluster-id')) {
```

Comment:
Rather than wrap everything in an if statement and duplicate sections, I think you could just have a variable that builds up the config and do the if statement for the sections that might be there and concatenate them if they are. 
e.g 
```
let config;
if (flag) {
  config += `${conditonal field || 'field'}`
}
...
```


Alternatively you could pull each section out to a variable and just have an if statement (or switch) and build the configs with the variables.
e.g.
```
const listener = `
listener "tcp" {
  ...
`

if (flag) {
  return `
  ${boundaryClusterId}
  ${listener}
  ${worker}
  ...
`
}
```